aries_cloudagent/settings/routes.py

```diff
@@ -0,0 +1,155 @@
+"""Settings routes."""
+
+import logging
+
+from aiohttp import web
+from aiohttp_apispec import docs, request_schema, response_schema
+from marshmallow import fields
+
+from ..admin.request_context import AdminRequestContext
+from ..multitenant.base import BaseMultitenantManager
+from ..core.error import BaseError
+from ..messaging.models.openapi import OpenAPISchema
+from ..multitenant.admin.routes import (
+    get_extra_settings_dict_per_tenant,
+    ACAPY_LIFECYCLE_CONFIG_FLAG_ARGS_MAP,
+)
+
+LOGGER = logging.getLogger(__name__)
+
+
+class UpdateProfileSettingsSchema(OpenAPISchema):
+    """Schema to update profile settings."""
+
+    extra_settings = fields.Dict(
+        description="Agent config key-value pairs",
+        required=False,
+        example={
+            "log_level": "INFO",
+            "ACAPY_INVITE_PUBLIC": True,
+            "public_invites": False,
+        },
+    )
+
+
+class ProfileSettingsSchema(OpenAPISchema):
+    """Profile settings response schema."""
+
+    settings = fields.Dict(
+        description="Profile settings dict",
+        example={
+            "log.level": "INFO",
+            "debug.invite_public": True,
+            "public_invites": False,
+        },
+    )
+
+
+def _get_filtered_settings_dict(wallet_settings: dict):
+    """Get filtered settings dict to display."""
+    filter_param_list = list(ACAPY_LIFECYCLE_CONFIG_FLAG_ARGS_MAP.values())
+    settings_dict = {}
+    for param in filter_param_list:
+        if param in wallet_settings:
+            settings_dict[param] = wallet_settings.get(param)
+    return settings_dict
+
+
+@docs(
+    tags=["settings"],
+    summary="Update settings or config associated with the profile.",
+)
+@request_schema(UpdateProfileSettingsSchema())
+@response_schema(ProfileSettingsSchema(), 200, description="")
+async def update_profile_settings(request: web.BaseRequest):
+    """
+    Request handler for updating setting associated with profile.
+
+    Args:
+        request: aiohttp request object
+    """
+    context: AdminRequestContext = request["context"]
+    root_profile = context.root_profile or context.profile
+    try:
+        body = await request.json()
+        extra_settings = get_extra_settings_dict_per_tenant(
+            body.get("extra_settings") or {}
+        )
+        async with root_profile.session() as session:
+            multitenant_mgr = session.inject_or(BaseMultitenantManager)
+            if multitenant_mgr:
+                wallet_id = context.metadata.get("wallet_id")
+                wallet_record = await multitenant_mgr.update_wallet(
+                    wallet_id, extra_settings
+                )
+                wallet_settings = wallet_record.settings
+                settings_dict = _get_filtered_settings_dict(wallet_settings)
```

Comment:
This is great, having the changes saved with the wallet settings. Only change is that the return of the update should match the return of the get. So return all of the profile settings not just the wallet settings. It should make it easier for the caller to know what all the settings are. For instance, they may cache these settings in their frontend code and returning all of the settings saves them from doing a merge...